components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java

```diff
@@ -20768,6 +20950,113 @@ public void updateAPIPoliciesMapping(String apiUUID, Set<URITemplate> uriTemplat
         }
     }
 
+    /**
+     * This method is to update Sequence Backend data
+     *
+     * @param apiUUID      API Id
+     * @param sequenceName Sequence Name
+     * @param sequence     Sequence Content
+     * @param type         Key type
+     * @param backendUUID  Sequence Id
+     * @throws APIManagementException If not properly updated
+     */
+    public void updateCustomBackend(String apiUUID, String sequenceName, String sequence, String type,
+            String backendUUID) throws APIManagementException {
+        // delete current working copy
+        String deleteCustomBackedQuery = SQLConstants.CustomBackendConstants.DELETE_CUSTOM_BACKEND_BY_API_AND_TYPE;
+        try (Connection connection = APIMgtDBUtil.getConnection();
+                PreparedStatement prepStmt = connection.prepareStatement(deleteCustomBackedQuery)) {
+            connection.setAutoCommit(false);
+            prepStmt.setString(1, apiUUID);
+            prepStmt.setString(2, type);
+            prepStmt.executeUpdate();
+            addCustomBackend(apiUUID, sequenceName, null, sequence, type, connection, backendUUID);
+            connection.commit();
+        } catch (SQLException e) {
+            handleException("Error while adding Custom Backend for API : " + apiUUID, e);
```

Comment:
We might need to add a rollback in case of an exception. Check other places as well.